rfcs/0007-service-date-filter.md

```diff
@@ -0,0 +1,104 @@
+| Carrier Claim                         | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+| DME (Durable Medical Equipment) Claim | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+| HHA (Home Health Agency) Claim        | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+| Hospice Claim                         | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+| Inpatient Claim                       | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+| Outpatient Claim                      | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+| PDE (Part D Event) Claim              | prescriptionFillDate | item#servicedDate  | SRVC_DT     |
+| SNF (Skilled Nursing Facility) Claim  | dateThrough          | billablePeriod#end | CLM_THRU_DT |
+
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+Currently, EOB lookup by patients accepts an optional date range filter (`_lastUpdated`). The lookup would be expanded to support an additional date range filter (`serviceEndDate`).
+
+The `serviceEndDate` filter will be implemented at the application layer, not the data layer. No schema changes are needed. After querying the database by patientId and (optionally) lastUpdated, the `serviceEndDate` filter checks the claim entity's date field against the provided date range. All claim entities that pass the filter are returned to the caller.
+
+If the caller does not supply `serviceEndDate`, no post query filtering occurs.
+
+Similar to the `_lastUpdated` parameter, the `serviceEndDate` is supplied as a date range. Users can supply a lower bound, upper bound, or lower/upper bound. The supported parameters include: `ge`, `gt`, `le`, `lt`.
+
+Sample Requests:
+* https://\<hostname\>/v1/fhir/ExplanationOfBenefit?patient=\<beneficiaryId\>&serviceEndDate=ge2020-01-01
+  * Returns claims data that occurred on or after 2020-01-01
+* https://\<hostname\>/v1/fhir/ExplanationOfBenefit?patient=\<beneficiaryId\>&serviceEndDate=lt2020-01-01
+  * Returns claims data that occurred before 2020-01-01
+* https://\<hostname\>/v1/fhir/ExplanationOfBenefit?patient=\<beneficiaryId\>&serviceEndDate=ge2020-01-01&serviceEndDate=le2020-01-31
+  * Returns claims data that occurred between 2020-01-01 and 2020-01-31 (inclusive)
+
+If a claim entity's date field is unset/null and the caller supplies a `serviceEndDate`, then the claim **will not** be returned to the caller. Since we cannot guarantee that the claim falls within the supplied filter, we cannot return it.
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+1. The `serviceEndDate` filter was previously implemented, but removed due to lack of adoption. While BCDA has identified a use case for this functionality, it may be BCDA-specific and thus should be implemented on the BCDA side.
```

Comment:
In general, BFD is interested in being the gatekeeper of data, so filtering on the BFD side makes sense for us.

Reply:
@mtrang1263 

**serviceEndDate** is not a EOB standard search param in FHIR R4 and CARIN. You can use HTTP header to pass serviceEndDate.

OR

Use the below CARIN defined EOB search param called service-date:

GET [base]/ExplanationOfBenefit?service-date=[service-date] and you can pass ExplanationOfBenefit.billablePeriod.end or ExplanationOfBenefit.item.servicedDate

Reply:
Didn't know there was a standard search param for service date, thanks for sharing @dshekhar18 ! 

Will update to use `service-date` instead.